### PR TITLE
Add hexagonal lattice CouplingMap

### PR DESCRIPTION
**Pull request by nahumsa**
### Summary

Adds `qiskit.transpiler.CouplingMap.from_hexagonal_lattice`.

### Details and comments

Add `qiskit.transpiler.CouplingMap.from_hexagonal_lattice`, which generates a hexagonal lattice as a coupling map.

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
This code looks correct to me, though do you have a particular use in mind for it?  There's a slight mistake in the release notes that are causing the documentation build to fail, but that's nothing major.

File: releasenotes/notes/add-hexagonal-lattice-couplingmap-d3b65b146b6cd1d1.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - |
+    Add a new constructor method, :meth: `~qiskit.transpiler.CouplingMap.from_hexagonal_lattice`, to
+    the :class:`~qiskit.transpiler.CouplingMap` class.
+
+    For example:
+
+    .. jupyter-execute::
+      from qiskit.transpiler import CouplingMap
+      cmap = CouplingMap.from_hexagonal_lattice(2, 2)
+      cmap.draw()
```
Comment (mtreinish) [inline releasenotes/notes/add-hexagonal-lattice-couplingmap-d3b65b146b6cd1d1.yaml:12]: This should fix the docs error:

```suggestion
    .. jupyter-execute::

      from qiskit.transpiler import CouplingMap
      cmap = CouplingMap.from_hexagonal_lattice(2, 2)
      cmap.draw()
```

**Comment by jakelishman** [comment]
Oh, that's weird - I made the same suggestion inline, but it never showed up in the review.  I must have forgotten to submit the comment...

**Comment by nahumsa** [comment]
> This code looks correct to me, though do you have a particular use in mind for it? There's a slight mistake in the release notes that are causing the documentation build to fail, but that's nothing major.

I think that it would be a good addition to the coupling maps, but I don't have any specific use in mind for it. :)

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for doing this. It seems straightforward enough and having an additional generator for playing around with device topologies easily locally doesn't hurt.

**Pull request merged**